rllib/algorithms/callbacks.py

```diff
@@ -51,13 +52,42 @@ def __init__(self, legacy_callbacks_dict: Dict[str, callable] = None):
             )
         self.legacy_callbacks = legacy_callbacks_dict or {}
 
+    @OverrideToImplementCustomLogic
+    def on_algorithm_init(
+        self,
+        *,
+        algorithm: "Algorithm",
+        **kwargs,
+    ) -> None:
+        """Callback run when a new algorithm instance has finished setup.
+
+        This method gets called at the end of Algorithm.setup() after all
+        the initialization is done, and before actually training starts.
+
+        Args:
+            algorithm: Reference to the trainer instance.
+            kwargs: Forward compatibility placeholder.
+        """
+        pass
+
+    @OverrideToImplementCustomLogic
+    def on_create_policy(self, *, policy_id: PolicyID, policy: Policy) -> None:
+        """Callback run whenever a new policy is added to an algorithm.
+
+        Args:
+            policy_id: ID of the newly created policy.
+            policy: the policy just created.
+        """
+        pass
+
     @OverrideToImplementCustomLogic
     def on_sub_environment_created(
         self,
         *,
         worker: "RolloutWorker",
         sub_environment: EnvType,
         env_context: EnvContext,
+        env_index: Optional[int] = None,
```

Comment:
This arg was already there and used by RLlib, just never added here to the signature!